import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -261,26 +283,67 @@ protected String getFormatFromBibLevel(Record record, char bibLevel, char format
      *
      * @param Record record
      * @param char recordType
+     * @param ControlField marc008
+     * @param List formatCodes007
      * @return String
      */
-    protected String getFormatFromRecordType(Record record, char recordType) {
+    protected String getFormatFromRecordType(Record record, char recordType, ControlField marc008, List formatCodes007) {
         switch (recordType) {
             case 'c':
             case 'd':
                 return "MusicalScore";
             case 'e':
             case 'f':
+                // Check 008/25 Type of cartographic material
+                switch (get008Value(marc008, 25)) {
+                    case 'd':
+                        return "Globe";
+                    case 'e':
+                        return "Atlas";
+                    default: break;
+                }
                 return "Map";
             case 'g':
-                // We're going to rely on the 007 instead for Projected Media
-                //return "Slide";
-                return "";
+                // Check 008/33 Type of visual material
+                switch (get008Value(marc008, 33)) {
+                    case 'f':
+                        return "Filmstrip";
+                    case 't':
+                        return "Transparency";
+                    case 'm':
+                        return "MotionPicture";
+                    case 'v': // Videorecording
+                        return "Video";
+                    default: break;
+                }
+                // Insufficient info in LDR and 008 to distinguish still from moving images
+                // If there is a 007 for either "Projected Graphic", "Motion Picture", or "Videorecording" 
+                // that should contain more information, so return nothing here.
+                // If no such 007 exists, fall back to "ProjectedMedium"
+                if (formatCodes007.contains('g') || formatCodes007.contains('m') || formatCodes007.contains('v')) {
```

Comment:
need to check for anything that could be a project medium type in 007, so 'g', 'm', 'v'. 

This should cover your first three examples, @demiankatz, i.e. the 008 has been coded as "other" ('z'), so we need to rely on one of the relevant 007s if they exist. Failing that, you'll end up with "ProjectedMedium" which is at least more accurate than "Slide"

Reply:
Thanks, @eocarragain! I'm going to repeat the indexing with the latest code and will report format counts later today!

Also, to answer your earlier question about commit history, don't worry about it -- I almost always just "squash and merge" to finish up pull requests, unless they are so complex that they justify a more granular commit history. I think this one should be fine to simply squash.

Reply:
this is done now too